controllers/btpoperator_controller.go

```diff
@@ -1343,6 +1348,43 @@ func (r *BtpOperatorReconciler) watchSecretPredicates() predicate.TypedPredicate
 	}
 }
 
+func (r *BtpOperatorReconciler) watchDeploymentPredicates() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			obj := e.Object.(*appsv1.Deployment)
+			return obj.Name == DeploymentName && obj.Namespace == ChartNamespace
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			obj := e.Object.(*appsv1.Deployment)
+			return obj.Name == DeploymentName && obj.Namespace == ChartNamespace
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			newObj := e.ObjectNew.(*appsv1.Deployment)
+			oldObj := e.ObjectOld.(*appsv1.Deployment)
+			if newObj.Name != DeploymentName || newObj.Namespace != ChartNamespace {
```

Comment:
```suggestion
			if !(newObj.Name == DeploymentName && newObj.Namespace == ChartNamespace) {
```

Reply:
That is how it was originally, but for me the other way round was more readable. We need a 3rd person for this ;)